<div>
<img src=https://www.institutedata.com/wp-content/uploads/2019/10/iod_h_tp_primary_c.svg width="300">
</div>

# Pandas Exploration

In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from scipy import stats
from seaborn import boxplot

# get matplot to show charts inline
%matplotlib inline

In [4]:
df = pd.read_csv("../DATA/Employee-attrition.csv")

In [5]:
df.head()

EmployeeID   recorddate_key birthdate_key orighiredate_key  \
0        1318  12/31/2006 0:00      1/3/1954        8/28/1989   
1        1318  12/31/2007 0:00      1/3/1954        8/28/1989   
2        1318  12/31/2008 0:00      1/3/1954        8/28/1989   
3        1318  12/31/2009 0:00      1/3/1954        8/28/1989   
4        1318  12/31/2010 0:00      1/3/1954        8/28/1989   

  terminationdate_key  age  length_of_service  city_name department_name  \
0            1/1/1900   52                 17  Vancouver       Executive   
1            1/1/1900   53                 18  Vancouver       Executive   
2            1/1/1900   54                 19  Vancouver       Executive   
3            1/1/1900   55                 20  Vancouver       Executive   
4            1/1/1900   56                 21  Vancouver       Executive   

  job_title  store_name gender_short gender_full termreason_desc  \
0       CEO          35            M        Male  Not Applicable   
1       CEO          35            M        Male  Not Applicable   
2       CEO          35            M        Male  Not Applicable   
3       CEO          35            M        Male  Not Applicable   
4       CEO          35            M        Male  Not Applicable   

    termtype_desc  STATUS_YEAR  STATUS BUSINESS_UNIT  
0  Not Applicable         2006  ACTIVE    HEADOFFICE  
1  Not Applicable         2007  ACTIVE    HEADOFFICE  
2  Not Applicable         2008  ACTIVE    HEADOFFICE  
3  Not Applicable         2009  ACTIVE    HEADOFFICE  
4  Not Applicable         2010  ACTIVE    HEADOFFICE

In [6]:
df.tail()

EmployeeID  recorddate_key birthdate_key orighiredate_key  \
49648        8258  12/1/2015 0:00     5/28/1994        8/19/2013   
49649        8264   8/1/2013 0:00     6/13/1994        8/27/2013   
49650        8279  12/1/2015 0:00     7/18/1994        9/15/2013   
49651        8296  12/1/2013 0:00      9/2/1994        10/9/2013   
49652        8321  12/1/2014 0:00    11/28/1994       11/24/2013   

      terminationdate_key  age  length_of_service    city_name  \
49648          12/30/2015   21                  2    Valemount   
49649           8/30/2013   19                  0    Vancouver   
49650          12/30/2015   21                  2   White Rock   
49651          12/31/2013   19                  0      Kelowna   
49652          12/30/2014   20                  1  Grand Forks   

        department_name     job_title  store_name gender_short gender_full  \
49648             Dairy  Dairy Person          34            M        Male   
49649  Customer Service       Cashier          44            F      Female   
49650  Customer Service       Cashier          39            F      Female   
49651  Customer Service       Cashier          16            F      Female   
49652  Customer Service       Cashier          13            F      Female   

      termreason_desc termtype_desc  STATUS_YEAR      STATUS BUSINESS_UNIT  
49648          Layoff   Involuntary         2015  TERMINATED        STORES  
49649      Resignaton     Voluntary         2013  TERMINATED        STORES  
49650          Layoff   Involuntary         2015  TERMINATED        STORES  
49651      Resignaton     Voluntary         2013  TERMINATED        STORES  
49652          Layoff   Involuntary         2014  TERMINATED        STORES

# Accessing the main DataFrame components

In [7]:
df.columns

Index(['EmployeeID', 'recorddate_key', 'birthdate_key', 'orighiredate_key',
       'terminationdate_key', 'age', 'length_of_service', 'city_name',
       'department_name', 'job_title', 'store_name', 'gender_short',
       'gender_full', 'termreason_desc', 'termtype_desc', 'STATUS_YEAR',
       'STATUS', 'BUSINESS_UNIT'],
      dtype='object')

In [8]:
df.index

RangeIndex(start=0, stop=49653, step=1)

In [9]:
df.values

array([[1318, '12/31/2006 0:00', '1/3/1954', ..., 2006, 'ACTIVE',
        'HEADOFFICE'],
       [1318, '12/31/2007 0:00', '1/3/1954', ..., 2007, 'ACTIVE',
        'HEADOFFICE'],
       [1318, '12/31/2008 0:00', '1/3/1954', ..., 2008, 'ACTIVE',
        'HEADOFFICE'],
       ...,
       [8279, '12/1/2015 0:00', '7/18/1994', ..., 2015, 'TERMINATED',
        'STORES'],
       [8296, '12/1/2013 0:00', '9/2/1994', ..., 2013, 'TERMINATED',
        'STORES'],
       [8321, '12/1/2014 0:00', '11/28/1994', ..., 2014, 'TERMINATED',
        'STORES']], dtype=object)

# 1. Understanding Data Types

#### 1.A Find Shape of Data Frame

```
# Expected Ouput
(49653, 18)
```

In [10]:
#ANSWER (1 line)
df.shape

(49653, 18)

#### 1.B Find Data Types of Data Frame
```
# Expected Ouput
EmployeeID              int64
recorddate_key         object
birthdate_key          object
...
dtype: object
```

In [11]:
#ANSWER (1 line)
df.dtypes

EmployeeID              int64
recorddate_key         object
birthdate_key          object
orighiredate_key       object
terminationdate_key    object
age                     int64
length_of_service       int64
city_name              object
department_name        object
job_title              object
store_name              int64
gender_short           object
gender_full            object
termreason_desc        object
termtype_desc          object
STATUS_YEAR             int64
STATUS                 object
BUSINESS_UNIT          object
dtype: object

#### 1.C Find Number of Data Types in a Data Frame
```
Expected Output:
int64      5
object    13
dtype: int64
```

In [12]:
#ANSWER (1 line) 
df.dtypes.value_counts(ascending = True)

int64      5
object    13
dtype: int64

#### 1.D Use .info()

Use .info() and explore difference.

In [13]:
#ANSWER (1 line)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49653 entries, 0 to 49652
Data columns (total 18 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   EmployeeID           49653 non-null  int64 
 1   recorddate_key       49653 non-null  object
 2   birthdate_key        49653 non-null  object
 3   orighiredate_key     49653 non-null  object
 4   terminationdate_key  49653 non-null  object
 5   age                  49653 non-null  int64 
 6   length_of_service    49653 non-null  int64 
 7   city_name            49653 non-null  object
 8   department_name      49653 non-null  object
 9   job_title            49653 non-null  object
 10  store_name           49653 non-null  int64 
 11  gender_short         49653 non-null  object
 12  gender_full          49653 non-null  object
 13  termreason_desc      49653 non-null  object
 14  termtype_desc        49653 non-null  object
 15  STATUS_YEAR          49653 non-null  int64 
 16  STAT

# 2. Selecting Column in Data Frame

#### 2.A Check Values of Job_Title column

```
# Expected Output
0                 CEO
1                 CEO
2                 CEO
...     
Name: job_title, Length: 49653, dtype: object
```

In [14]:
#ANSWER (1 line)
df['job_title']

0                 CEO
1                 CEO
2                 CEO
3                 CEO
4                 CEO
             ...     
49648    Dairy Person
49649         Cashier
49650         Cashier
49651         Cashier
49652         Cashier
Name: job_title, Length: 49653, dtype: object

#### 2.B Find How Many Rows Are in Each Job Title



```
# Expected Output
Meat Cutter                        9984
Dairy Person                       8590
Produce Clerk                      8237
...
Name: job_title, dtype: int64
```



In [15]:
#ANSWER (1 line)
df['job_title'].value_counts()

Meat Cutter                        9984
Dairy Person                       8590
Produce Clerk                      8237
Baker                              8096
Cashier                            6816
Shelf Stocker                      5622
Customer Service Manager            306
Processed Foods Manager             289
Meats Manager                       285
Bakery Manager                      285
Produce Manager                     278
Store Manager                       271
Recruiter                            62
HRIS Analyst                         55
Accounting Clerk                     50
Benefits Admin                       35
Accounts Receiveable Clerk           30
Labor Relations Analyst              30
Trainer                              26
Accounts Payable Clerk               25
Investment Analyst                   20
Compensation Analyst                 20
Auditor                              20
Systems Analyst                      20
Corporate Lawyer                     17


#### 2.C [BONUS] Write A Function to Row Count

In [28]:
def row_count(column_name):
    number_of_rows = column_name.shape
    return number_of_rows

row_count(df['recorddate_key'])

(49653,)

#### 2.D Describe 'Age'

Find count,  mean,  std, min, max of column 'age'

```
Expected Output:
count    49653.000000
mean        42.077035
std         12.427257
...
Name: age, dtype: float64
```

In [17]:
#ANSWER (1 line)

df['age'].describe()

count    49653.000000
mean        42.077035
std         12.427257
min         19.000000
25%         31.000000
50%         42.000000
75%         53.000000
max         65.000000
Name: age, dtype: float64

##### 2.E [BONUS] Describe 'All Numeric Columns of Data Frame

Find count, mean, std, min, max of of all columns of dataframe

```
Expected Output:
```
||EmployeeID|age|length_of_service|store_name|STATUS_YEAR|
|:--|:--|:--|:--|:--|:--|
|**count**|49653.000000|49653.000000|49653.000000|49653.000000|49653.000000|
|**mean**|4859.495740|42.077035|10.43459|27.297605|2010.612612|
|...|...|...|...|...|

In [18]:
#ANSWER (1 line)
df.describe()

EmployeeID           age  length_of_service    store_name  \
count  49653.000000  49653.000000       49653.000000  49653.000000   
mean    4859.495740     42.077035          10.434596     27.297605   
std     1826.571142     12.427257           6.325286     13.514134   
min     1318.000000     19.000000           0.000000      1.000000   
25%     3360.000000     31.000000           5.000000     16.000000   
50%     5031.000000     42.000000          10.000000     28.000000   
75%     6335.000000     53.000000          15.000000     42.000000   
max     8336.000000     65.000000          26.000000     46.000000   

        STATUS_YEAR  
count  49653.000000  
mean    2010.612612  
std        2.845577  
min     2006.000000  
25%     2008.000000  
50%     2011.000000  
75%     2013.000000  
max     2015.000000

# 3. Making The Index Meaningful
**Set the Dataframe index to the Employee ID Column**
```
No Expected Output
```

In [19]:
#ANSWER (1 line)
df.set_index('EmployeeID', inplace = True)

#### 3.A Find A Specifc Row Using Index
```
Expected Output:
recorddate_key         ??/??/???? ??:??
birthdate_key                ??/??/????
orighiredate_key             ??/??/????
...
Name: ???, dtype: object
```

In [20]:
#ANSWER (1 line)

df.iloc[300]

recorddate_key         12/31/2011 0:00
birthdate_key                4/20/1949
orighiredate_key            10/24/1989
terminationdate_key          4/20/2014
age                                 62
length_of_service                   22
city_name                    Vancouver
department_name            Recruitment
job_title                    Recruiter
store_name                          35
gender_short                         F
gender_full                     Female
termreason_desc         Not Applicable
termtype_desc           Not Applicable
STATUS_YEAR                       2011
STATUS                          ACTIVE
BUSINESS_UNIT               HEADOFFICE
Name: 1382, dtype: object

#### 3.B Sort Data Frame using Index
```
Expected Output:
```
|EmployeeID|recorddate_key|birthdate_key|orighiredate_key|terminationdate_key|age|...|
|:--|:--|:--|:--|:--|:--|:--|
|**1318**|12/31/2006 0:00|1/3/1954|8/28/1989|1/1/1900|52|...|
|**1318**|12/31/2007 0:00|1/3/1954|8/28/1989|1/1/1900|53|...|
|...|...|...|...|...|...|...|

In [21]:
#ANSWER (1 line)
df.sort_index()

recorddate_key birthdate_key orighiredate_key  \
EmployeeID                                                   
1318        12/31/2006 0:00      1/3/1954        8/28/1989   
1318        12/31/2007 0:00      1/3/1954        8/28/1989   
1318        12/31/2008 0:00      1/3/1954        8/28/1989   
1318        12/31/2009 0:00      1/3/1954        8/28/1989   
1318        12/31/2010 0:00      1/3/1954        8/28/1989   
...                     ...           ...              ...   
8335        12/31/2013 0:00    12/28/1994       12/10/2013   
8335        12/31/2015 0:00    12/28/1994       12/10/2013   
8336        12/31/2015 0:00    12/31/1994       12/11/2013   
8336        12/31/2013 0:00    12/31/1994       12/11/2013   
8336        12/31/2014 0:00    12/31/1994       12/11/2013   

           terminationdate_key  age  length_of_service  city_name  \
EmployeeID                                                          
1318                  1/1/1900   52                 17  Vancouver   
1318                  1/1/1900   53                 18  Vancouver   
1318                  1/1/1900   54                 19  Vancouver   
1318                  1/1/1900   55                 20  Vancouver   
1318                  1/1/1900   56                 21  Vancouver   
...                        ...  ...                ...        ...   
8335                  1/1/1900   19                  0  Vancouver   
8335                  1/1/1900   21                  2  Vancouver   
8336                  1/1/1900   21                  2  Vancouver   
8336                  1/1/1900   19                  0  Vancouver   
8336                  1/1/1900   20                  1  Vancouver   

           department_name     job_title  store_name gender_short gender_full  \
EmployeeID                                                                      
1318             Executive           CEO          35            M        Male   
1318             Executive           CEO          35            M        Male   
1318             Executive           CEO          35            M        Male   
1318             Executive           CEO          35            M        Male   
1318             Executive           CEO          35            M        Male   
...                    ...           ...         ...          ...         ...   
8335                 Dairy  Dairy Person          35            F      Female   
8335                 Dairy  Dairy Person          35            F      Female   
8336                 Dairy  Dairy Person          35            M        Male   
8336                 Dairy  Dairy Person          35            M        Male   
8336                 Dairy  Dairy Person          35            M        Male   

           termreason_desc   termtype_desc  STATUS_YEAR  STATUS BUSINESS_UNIT  
EmployeeID                                                                     
1318        Not Applicable  Not Applicable         2006  ACTIVE    HEADOFFICE  
1318        Not Applicable  Not Applicable         2007  ACTIVE    HEADOFFICE  
1318        Not Applicable  Not Applicable         2008  ACTIVE    HEADOFFICE  
1318        Not Applicable  Not Applicable         2009  ACTIVE    HEADOFFICE  
1318        Not Applicable  Not Applicable         2010  ACTIVE    HEADOFFICE  
...                    ...             ...          ...     ...           ...  
8335        Not Applicable  Not Applicable         2013  ACTIVE        STORES  
8335        Not Applicable  Not Applicable         2015  ACTIVE        STORES  
8336        Not Applicable  Not Applicable         2015  ACTIVE        STORES  
8336        Not Applicable  Not Applicable         2013  ACTIVE        STORES  
8336        Not Applicable  Not Applicable         2014  ACTIVE        STORES  

[49653 rows x 17 columns]

#### 3.C [BONUS] Sort Data Frame using Age
```
Expected Output:
```
|EmployeeID|recorddate_key|birthdate_key|orighiredate_key|terminationdate_key|age|...|
|:--|:--|:--|:--|:--|:--|:--|
|**8325**|12/31/2013 0:00|12/7/1994|11/29/2013|1/1/1900|19|...|
|**8253**|12/31/2013 0:00|5/15/1994|8/12/2013|1/1/1900|19|...|
|...|...|...|...|...|...|...|

In [22]:
#ANSWER (1 line)
df.sort_values('age')


recorddate_key birthdate_key orighiredate_key  \
EmployeeID                                                   
8325        12/31/2013 0:00     12/7/1994       11/29/2013   
8253        12/31/2013 0:00     5/15/1994        8/12/2013   
8252        12/31/2013 0:00     5/10/1994         8/9/2013   
8251        12/31/2013 0:00      5/7/1994         8/7/2013   
8250        12/31/2013 0:00     4/26/1994         8/2/2013   
...                     ...           ...              ...   
3262          8/1/2008 0:00     8/23/1943        2/28/1995   
3248          8/1/2008 0:00     8/14/1943        2/19/1995   
4948          7/1/2013 0:00      7/9/1948        1/14/2000   
3356         11/1/2008 0:00    11/18/1943        5/26/1995   
4138          2/1/2011 0:00     2/12/1946        8/20/1997   

           terminationdate_key  age  length_of_service        city_name  \
EmployeeID                                                                
8325                  1/1/1900   19                  0        Cranbrook   
8253                  1/1/1900   19                  0  North Vancouver   
8252                  1/1/1900   19                  0         Victoria   
8251                  1/1/1900   19                  0        Vancouver   
8250                  1/1/1900   19                  0          Burnaby   
...                        ...  ...                ...              ...   
3262                 8/23/2008   65                 13          Nanaimo   
3248                 8/14/2008   65                 13        Vancouver   
4948                  7/9/2013   65                 13          Burnaby   
3356                11/18/2008   65                 13          Nanaimo   
4138                 2/12/2011   65                 13           Surrey   

             department_name      job_title  store_name gender_short  \
EmployeeID                                                             
8325        Customer Service        Cashier           8            M   
8253        Customer Service        Cashier          22            F   
8252        Customer Service        Cashier          46            F   
8251        Customer Service        Cashier          44            F   
8250                   Dairy   Dairy Person           5            F   
...                      ...            ...         ...          ...   
3262                   Meats    Meat Cutter          18            F   
3248                   Dairy   Dairy Person          35            F   
4948                 Produce  Produce Clerk           5            F   
3356                   Meats    Meat Cutter          18            F   
4138                 Produce  Produce Clerk          31            F   

           gender_full termreason_desc   termtype_desc  STATUS_YEAR  \
EmployeeID                                                            
8325              Male  Not Applicable  Not Applicable         2013   
8253            Female  Not Applicable  Not Applicable         2013   
8252            Female  Not Applicable  Not Applicable         2013   
8251            Female  Not Applicable  Not Applicable         2013   
8250            Female  Not Applicable  Not Applicable         2013   
...                ...             ...             ...          ...   
3262            Female      Retirement       Voluntary         2008   
3248            Female      Retirement       Voluntary         2008   
4948            Female      Retirement       Voluntary         2013   
3356            Female      Retirement       Voluntary         2008   
4138            Female      Retirement       Voluntary         2011   

                STATUS BUSINESS_UNIT  
EmployeeID                            
8325            ACTIVE        STORES  
8253            ACTIVE        STORES  
8252            ACTIVE        STORES  
8251            ACTIVE        STORES  
8250            ACTIVE        STORES  
...                ...           ...  
3262        TERMINATED        STORES  
3248        TERMIN

#### 3.D [BONUS] Using Sample Select Any Row 
```
Expected Output:
```
|EmployeeID|recorddate_key|birthdate_key|orighiredate_key|terminationdate_key|age|...|
|:--|:--|:--|:--|:--|:--|:--|
|**????**|??/??/???? ??:??|??/??/????|??/??/????|??/??/????|??|...|
|...|...|...|...|...|...|...|

In [23]:
#ANSWER (1 line)

df.iloc[1348:4138]

recorddate_key birthdate_key orighiredate_key  \
EmployeeID                                                   
1647        12/31/2008 0:00     9/30/1950        7/30/1990   
1647        12/31/2009 0:00     9/30/1950        7/30/1990   
1650        12/31/2006 0:00     10/5/1950         8/2/1990   
1650        12/31/2007 0:00     10/5/1950         8/2/1990   
1650        12/31/2008 0:00     10/5/1950         8/2/1990   
...                     ...           ...              ...   
2122        12/31/2012 0:00     6/21/1953         1/6/1992   
2122        12/31/2013 0:00     6/21/1953         1/6/1992   
2122        12/31/2014 0:00     6/21/1953         1/6/1992   
2122        12/31/2015 0:00     6/21/1953         1/6/1992   
2123        12/31/2006 0:00     6/23/1953         1/7/1992   

           terminationdate_key  age  length_of_service        city_name  \
EmployeeID                                                                
1647                 9/30/2010   58                 18    Prince George   
1647                 9/30/2010   59                 19    Prince George   
1650                12/30/2015   56                 16         Victoria   
1650                12/30/2015   57                 17         Victoria   
1650                12/30/2015   58                 18         Victoria   
...                        ...  ...                ...              ...   
2122                  1/1/1900   59                 20        Vancouver   
2122                  1/1/1900   60                 21        Vancouver   
2122                  1/1/1900   61                 22        Vancouver   
2122                  1/1/1900   62                 23        Vancouver   
2123                  1/1/1900   53                 14  New Westminster   

           department_name      job_title  store_name gender_short  \
EmployeeID                                                           
1647                 Dairy   Dairy Person          26            M   
1647                 Dairy   Dairy Person          26            M   
1650                 Meats    Meat Cutter          37            F   
1650                 Meats    Meat Cutter          37            F   
1650                 Meats    Meat Cutter          37            F   
...                    ...            ...         ...          ...   
2122               Produce  Produce Clerk          41            M   
2122               Produce  Produce Clerk          41            M   
2122               Produce  Produce Clerk          41            M   
2122               Produce  Produce Clerk          41            M   
2123                 Meats    Meat Cutter          21            F   

           gender_full termreason_desc   termtype_desc  STATUS_YEAR  STATUS  \
EmployeeID                                                                    
1647              Male  Not Applicable  Not Applicable         2008  ACTIVE   
1647              Male  Not Applicable  Not Applicable         2009  ACTIVE   
1650            Female  Not Applicable  Not Applicable         2006  ACTIVE   
1650            Female  Not Applicable  Not Applicable         2007  ACTIVE   
1650            Female  Not Applicable  Not Applicable         2008  ACTIVE   
...                ...             ...             ...          ...     ...   
2122              Male  Not Applicable  Not Applicable         2012  ACTIVE   
2122              Male  Not Applicable  Not Applicable         2013  ACTIVE   
2122              Male  Not Applicable  Not Applicable         2014  ACTIVE   
2122              Male  Not Applicable  Not Applicable         2015  ACTIVE   
2123            Female  Not Applicable  Not Applicable         2006  ACTIVE   

           BUSINESS_UNIT  
EmployeeID                
1647              STORES  
1647              STORES  
1650              STORES  
1650              STORES  
1650              STORES  
...                  ...  
2122              STORES  
2122              STORES  
2122              STO

# 4. Query Data Frame

#### 4.A Find Rows Where Job_Title is CEO
```
Expected Output:
```
|EmployeeID|recorddate_key|birthdate_key|orighiredate_key|terminationdate_key|age|...|
|:--|:--|:--|:--|:--|:--|:--|
|**1318**|12/31/2006 0:00|1/3/1954|8/28/1989|1/1/1900|52|...|
|**1318**|12/31/2007 0:00|1/3/1954|8/28/1989|1/1/1900|53|...|
|...|...|...|...|...|...|...|

In [24]:
#ANSWER (1 line)
df[df.job_title== 'CEO']

recorddate_key birthdate_key orighiredate_key  \
EmployeeID                                                   
1318        12/31/2006 0:00      1/3/1954        8/28/1989   
1318        12/31/2007 0:00      1/3/1954        8/28/1989   
1318        12/31/2008 0:00      1/3/1954        8/28/1989   
1318        12/31/2009 0:00      1/3/1954        8/28/1989   
1318        12/31/2010 0:00      1/3/1954        8/28/1989   
1318        12/31/2011 0:00      1/3/1954        8/28/1989   
1318        12/31/2012 0:00      1/3/1954        8/28/1989   
1318        12/31/2013 0:00      1/3/1954        8/28/1989   
1318        12/31/2014 0:00      1/3/1954        8/28/1989   
1318        12/31/2015 0:00      1/3/1954        8/28/1989   

           terminationdate_key  age  length_of_service  city_name  \
EmployeeID                                                          
1318                  1/1/1900   52                 17  Vancouver   
1318                  1/1/1900   53                 18  Vancouver   
1318                  1/1/1900   54                 19  Vancouver   
1318                  1/1/1900   55                 20  Vancouver   
1318                  1/1/1900   56                 21  Vancouver   
1318                  1/1/1900   57                 22  Vancouver   
1318                  1/1/1900   58                 23  Vancouver   
1318                  1/1/1900   59                 24  Vancouver   
1318                  1/1/1900   60                 25  Vancouver   
1318                  1/1/1900   61                 26  Vancouver   

           department_name job_title  store_name gender_short gender_full  \
EmployeeID                                                                  
1318             Executive       CEO          35            M        Male   
1318             Executive       CEO          35            M        Male   
1318             Executive       CEO          35            M        Male   
1318             Executive       CEO          35            M        Male   
1318             Executive       CEO          35            M        Male   
1318             Executive       CEO          35            M        Male   
1318             Executive       CEO          35            M        Male   
1318             Executive       CEO          35            M        Male   
1318             Executive       CEO          35            M        Male   
1318             Executive       CEO          35            M        Male   

           termreason_desc   termtype_desc  STATUS_YEAR  STATUS BUSINESS_UNIT  
EmployeeID                                                                     
1318        Not Applicable  Not Applicable         2006  ACTIVE    HEADOFFICE  
1318        Not Applicable  Not Applicable         2007  ACTIVE    HEADOFFICE  
1318        Not Applicable  Not Applicable         2008  ACTIVE    HEADOFFICE  
1318        Not Applicable  Not Applicable         2009  ACTIVE    HEADOFFICE  
1318        Not Applicable  Not Applicable         2010  ACTIVE    HEADOFFICE  
1318        Not Applicable  Not Applicable         2011  ACTIVE    HEADOFFICE  
1318        Not Applicable  Not Applicable         2012  ACTIVE    HEADOFFICE  
1318        Not Applicable  Not Applicable         2013  ACTIVE    HEADOFFICE  
1318        Not Applicable  Not Applicable         2014  ACTIVE    HEADOFFICE  
1318        Not Applicable  Not Applicable         2015  ACTIVE    HEADOFFICE

#### 4.B Group By Age
```
Expected Output:
age
19     158
20     408
21     703
...
dtype: int64
```

In [25]:
#ANSWER (1 line)
df.groupby('age').size()

age
19     158
20     408
21     703
22     815
23     960
24    1111
25    1197
26    1210
27    1235
28    1225
29    1227
30    1212
31    1146
32    1153
33    1164
34    1188
35    1189
36    1176
37    1149
38    1156
39    1142
40    1130
41    1135
42    1152
43    1150
44    1157
45    1141
46    1161
47    1173
48    1180
49    1196
50    1218
51    1207
52    1188
53    1188
54    1159
55    1168
56    1154
57    1130
58    1130
59    1128
60    1109
61     757
62     712
63     667
64     646
65     593
dtype: int64

# 5. Visualization
#### 5.A Plot Histogram of Age 
```
Desired Output:

```

In [ ]:
"""CODE DELETED. DO NOT TOUCH THIS TEXT. RUNNING THIS CELL WITH SHIFT-ENTER WILL DELETE THE GRAPH"""

In [ ]:
#ANSWER (5 lines)

#### 5.B BoxPlot of Age
```
Desired Output:

```

In [ ]:
"""CODE DELETED. DO NOT TOUCH THIS TEXT. RUNNING THIS CELL WITH SHIFT-ENTER WILL DELETE THE GRAPH"""

In [ ]:
#ANSWER (5 lines)
boxplot(y=???)

#### 5.C Scatterplot of age and length_of_service
```
Desired Output:

```

In [ ]:
"""CODE DELETED. DO NOT TOUCH THIS TEXT. RUNNING THIS CELL WITH SHIFT-ENTER WILL DELETE THE GRAPH"""

In [ ]:
#ANSWER (5 lines)

#### 5.D Plot Histogram of store_name
```
Desired Output:

```

In [ ]:
"""CODE DELETED. DO NOT TOUCH THIS TEXT. RUNNING THIS CELL WITH SHIFT-ENTER WILL DELETE THE GRAPH"""

In [ ]:
#ANSWER (5 lines)

#### 5.E Print Statistical Summary of Age using the stats library

In [ ]:
#ANSWER (1 line)



---



---



> > > > > > > > > © 2021 Institute of Data


---



---



